common/src/worldgen.rs

```diff
@@ -340,6 +320,82 @@ impl ChunkParams {
         criteria_met >= 2
     }
 
+    fn voxel_neighbors(&self, coords: na::Vector3<u8>, voxels: &VoxelData) -> [NeighborData; 6] {
+        [
+            self.neighbor(coords, -1, 0, 0, &voxels),
+            self.neighbor(coords, 1, 0, 0, &voxels),
+            self.neighbor(coords, 0, -1, 0, &voxels),
+            self.neighbor(coords, 0, 1, 0, &voxels),
+            self.neighbor(coords, 0, 0, -1, &voxels),
+            self.neighbor(coords, 0, 0, 1, &voxels),
+        ]
+    }
+
+    fn neighbor(
+        &self,
```

Comment:
Why was this made a method? It doesn't seem to need access to `self`.

Reply:
I assume that with this and the other question, you are asking why they are methods as opposed to functions? There's no particular reason they need to be methods. The only part of `self` that they collectively needed was `self.dimension`. I've made the chunk size an additional argument of both functions and moved them out of `impl ChunkParams`.

Reply:
Whoops, I didn't actually see `dimension` on them; it was late. That's a fine reason for them to be methods; feel free to roll back that last commit.

Reply:
It's done, also rebased on top of the other PR that you closed.